acs-engine/preparation/script.md

```diff
@@ -157,7 +157,29 @@ Retrying role assignment creation: 1/36
   "tenant": "72f988bf-xxxx-xxxx-xxxx-2d7cd011db47"
 }
 ```
+## If using an existing service princpal
+If a service principal already exists, the command below will assign permissions to the Resource Group created for ACSE.  In some environments, the Service Principal may need to be created out of band due to Azure AD security policies which prevent non-admins from creating Service Principals.  Note that --assignee is the appId instead of the name.
 
 
+```
+az role assignment create --assignee "657cd27f-xxxx-xxxx-xxxx-512adb26e468"  --role Contributor --resource-group $ACSE_RESOURCE_GROUP
```

Comment:
This will break the "test" mode of SimDem as the assignee is not a real assignee. It is very important that all scripts pass tests as we will eventually be enabling CI on all these scripts.

The goal of these scripts is not to document every possible flow. It is to document a single scenario that can then be reliably and predictably reproduced. It should be the goal of documentation to describe all possible flows.

I do agree having a hint in this document would be useful since the goal is for these to be reused within documentation too. I see at least three ways of handling this...

    1. Create a separate script to describe the process for a pre-existing SP (seems like this case is too trivial for a separate demo)
    2. Keep the instructions here, but don't make them executable content (i.e. not in a code block, but in an inline, single backtick, code snippet within the explanatory paragraph)
    3. Link out to the CLI documentation on this topic